app/components/primer/alpha/segmented_control/item.rb

```diff
@@ -13,11 +13,23 @@ class Item < Primer::BaseComponent
         # @param selected [Boolean] Whether the item is selected
         # @param icon [Symbol] The icon to use
         # @param hide_labels [Symbol] Whether to only show the icon
-        def initialize(label:, selected: false, icon: nil, hide_labels: false, **system_arguments)
+        # @param trailing_visual_label [String] A trailing visual label that appears to the right of the item
+        # @param trailing_visual_label_options [Hash] Same arguments as <%= link_to_component(Primer::Beta::Label) %>
+        def initialize(
+          label:,
+          selected: false,
+          icon: nil,
+          hide_labels: false,
+          trailing_visual_label: nil,
+          trailing_visual_label_options: {},
+          **system_arguments
+        )
           @icon = icon
           @hide_labels = hide_labels
           @label = label
           @selected = selected
+          @trailing_visual_label = trailing_visual_label
+          @trailing_visual_label_options = trailing_visual_label_options
```

Comment:
Instead of passing these into the initializer, we can keep it cleaner by creating a `with_trailing_visual_label` def in the class. If we create the button in the initializer, we'll be able to add a slot instance with a def. An example were we do this already is in the action menu 

This creates the `@list` object in the inilializer
https://github.com/primer/view_components/blob/d92acd927c6ab559a834632cb7b60a972ad5b154/app/components/primer/alpha/action_menu.rb#L77-L81

This adds to the object slot https://github.com/primer/view_components/blob/d92acd927c6ab559a834632cb7b60a972ad5b154/app/components/primer/alpha/action_menu.rb#L103-L108

Then it gets rendered in the view like this https://github.com/primer/view_components/blob/d92acd927c6ab559a834632cb7b60a972ad5b154/app/components/primer/alpha/action_menu.html.erb#L22

Reply:
@jonrohan I tried this in https://github.com/primer/view_components/pull/2278/commits/90e7ada50bc0b0644caaf11ef16ec138a8c60376, but I must be missing something, since I'm not having luck with getting the label to appear in lookbook.

You're welcome to commit directly to the branch too, if that's easier.

Reply:
I can't really explain why this works but rendering the content before render fixes it https://github.com/primer/view_components/pull/2278/commits/dc2a145da0f5b666ff049da51e6dd75089d9f062

Reply:
Interesting, without the `before_render`, I think the nested notation fails, but if we chained, it produced the expected content.

For example:
```ruby
component.with_item(label: "Yearly", selected: true).with_trailing_visual_label(scheme: :accent) { "-8%" }
```

But not:
```ruby
component.with_item(label: "Yearly", selected: true) do |button|
  button.with_trailing_visual_label(scheme: :accent) { "-8%" }
end
```

Likely, something due render order.